charts/gardener/controlplane/charts/runtime/templates/admission-controller/deployment.yaml

```diff
@@ -0,0 +1,93 @@
+{{- if .Values.global.admission.enabled }}
+apiVersion: {{ include "deploymentversion" . }}
+kind: Deployment
+metadata:
+  name: gardener-admission-controller
+  namespace: garden
+  labels:
+    app: gardener
+    role: admission-controller
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: {{ required ".Values.global.admission.replicaCount is required" .Values.global.admission.replicaCount }}
+  selector:
+    matchLabels:
+      app: gardener
+      role: admission-controller
+  template:
+    metadata:
+      annotations:
+        checksum/configmap-gardener-admission-controller-config: {{ include (print $.Template.BasePath "/admission-controller/configmap-componentconfig.yaml") . | sha256sum }}
+        checksum/secret-admission-controller-cert: {{ include (print $.Template.BasePath "/admission-controller/secret-cert.yaml") . | sha256sum }}
+        checksum/secret-gardener-admission-controller-kubeconfig: {{ include (print $.Template.BasePath "/admission-controller/secret-kubeconfig.yaml") . | sha256sum }}
+        {{- if .Values.global.admission.podAnnotations }}
+{{ toYaml .Values.global.admission.podAnnotations | indent 8 }}
+        {{- end }}
+      labels:
+        app: gardener
+        role: admission-controller
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+        {{- if .Values.global.admission.podLabels }}
+{{ toYaml .Values.global.admission.podLabels | indent 8 }}
+        {{- end }}
+    spec:
+      {{- if not .Values.global.admission.kubeconfig }}
+      serviceAccountName: {{ required ".Values.global.admission.serviceAccountName is required" .Values.global.admission.serviceAccountName }}
+      {{- end }}
+      {{- if gt (int .Values.global.admission.replicaCountreplicaCount) 1 }}
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: app
+                operator: In
+                values:
+                - gardener
+              - key: role
+                operator: In
+                values:
+                - admission-controller
+            topologyKey: kubernetes.io/hostname
+      {{- end }}
+      containers:
+      - name: gardener-admission-controller
+        image: {{ include "utils-templates.image" .Values.global.admission.image }}
+        imagePullPolicy: {{ .Values.global.admission.image.pullPolicy }}
+        command:
+        - /gardener-admission-controller
+        - --config=/etc/gardener-admission-controller/config/config.yaml
+        {{- if .Values.global.admission.resources }}
+        resources:
+{{ toYaml .Values.global.admission.resources | indent 10 }}
+        {{- end }}
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
```

Comment:
Can you add some health checks? I think we should at least add a TCP liveness and readiness probe.

Reply:
See discussion about health checks [here](https://github.com/gardener/gardener-extension-provider-gcp/pull/38#discussion_r387523877) and why we did not add them yet to any webhook component. Readiness is fine since we now wait for the cache sync.